add time/date editor, other menu fixes

EJ125 opened the pull request:
added ability to edit time and date, and moved display of time/date/moonphase info into tools menu

EJ125 committed:
adding time/date editor, moving time/day info display

EJ125 committed:
adding time/date editor, moving time/day info display

EJ125 committed:
adding time/date editor, moving time/day info display

EJ125 committed:
adding time/date editor, moving time/day info display

EJ125 committed:
Add files via upload

TakaRikka commented on source/menus/tools_menu.cpp:
```diff
@@ -43,6 +49,33 @@ Line lines[LINE_NUM] = {
+
+    tww_sprintf(Time, "%02d:%02d", hour, min);
+    tww_sprintf(Date, "date: %d", date);
+    tww_sprintf(Moon, "Moon: %d", moonphases[0]);
```
this doesn't seem right, should be `tww_sprintf(Moon, "Moon: %s", moonphases[date % 7]);` i think?

EJ125 committed:
Update tools_menu.cpp

remove the useless sent of tww_spfrintf

EJ125 committed:
fixing dpad left/right bug for menus that only have A/B press interaction

EJ125 committed:
fix tuner flag menu draw issue

this fixes the weird scroll bug, which is because with how the menus for flags are written, the game thinks it is above MAX_RENDER_:LINES even though there are only 4 lines. so I increased the max enough to hide the issue, causes no overlap of words or words cut-off screen in any menu, but it pretty close, so if we wanted to add more flags we would need to actually fix the issue

EJ125 committed:
resolving conflicts with libfix's

EJ125 committed:
resolcing conflicts with libfix
